### ISSUE TITLE
Accessibility in HighCharts stock

### ISSUE THREAD

**Issue by davidnm** [highcharts/highcharts#20267]
We are trying to get a highchart-stocks chart fully accesible for both, **keyboard **and voiceover

We have detected that the HighStock stock-tools module does not correctly add the necessary elements in the DOM generated so that accessibility is not persistent in the HTML structure, for example, in the list generated, it does not incorporate the tabindex attribute in the same way as it is added when using the exporting export module:

HTML for exporting:
![image](https://github.com/highcharts/highcharts/assets/8101446/fbcd6723-2d20-4ab2-b4ef-41b54a84341c)

stocktools HTML: 
![image](https://github.com/highcharts/highcharts/assets/8101446/d52e601b-2ff0-43f3-8556-aeb6efcd91f8)

As you can see, the tabindex attribute is not included in the second case.

On the other hand, semantically it is not being constructed correctly for the elements within each “li”. This does not allow correct use of the “voiceover” option.

It would be interesting if the structure were more semantic with “buttons”, “tabindex” and “aria-label” or “role” where necessary within each module, or at least within the stocktools module.

Keyboard access to the popup is also not allowed, which affects the user's usability experience.

Could you please let us know if we are missing something or there are any other modules to use for accesibility related to highchart stocks charts?

thank you in advanced

**Comment by alu-ms** [comment]
Hi @pawelfus -
Our team encountered a similar issue as well. 
We leveraged the click event with a customized function, which is calling out a side panel. But looks like is not included in accessibility yet. I was able to navigate, but cannot find where to pass on my customized function for keyboard actions.
More details:
`Highcharts.Options=plotOptions=>series=>point=>events=>click`
In click, that's where we were able to pass our customized function, but did not find where to do so for keyboard actions.

Looking forward to this feature to be rolled out.

Thanks!

**Comment by hubertkozik** [comment]
Hi @alu-ms! It seems, that this issue is covered in https://github.com/highcharts/highcharts/pull/20386. Due to that, I am closing this ticket as completed.

**Comment by hubertkozik** [comment]
@alu-ms If you need any help with Highcharts configuration, feel free to reach out to our other [support channels ](https://www.highcharts.com/blog/support/), where our support engineers will help you with the configuration of your chart.

**Comment by alu-ms** [comment]
Hi @hubertkozik  - thanks for getting back! I created a new question here: https://stackoverflow.com/questions/78071398/highcharts-accessibility-enter-and-click-events-function-mouseover-is-not-enoug.

**Comment by jedrzejruta** [comment]
Reopening, although this isn't exactly the same issue as described in the task description, the problem with a11y still exists - the Accessibility component doesn't include Stock Tools, meaning it's not accessible by keyboard or screen readers.

As a workaround, users can destroy and init again the Accessibility component after the chart is loaded:
```javascript
chart.accessibility.destroy();
chart.accessibility.init(chart);
``` 

**Workaround demo:** https://jsfiddle.net/BlackLabel/6mcjs837/
